tests/test_calendar_analyzer.py

```
@@ -138,19 +156,34 @@ def test_valid_date_formats(monkeypatch, capsys):
 def test_edge_case_dates(monkeypatch, capsys):
     """Test edge case date formats."""
     # Test leap year date
+    # Create a temporary dummy file path that doesn't exist
+    import tempfile
+    dummy_file = tempfile.NamedTemporaryFile(suffix=".ics", delete=False)
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)